### PR TITLE
pesum - add a new tool to the shed

### PR DESCRIPTION
**Pull request by vathpela**
This PR adds a new tool, "pesum", that works like such:
```
random:~/devel/github.com/pesign/pesum$ ./src/pesum shimx64.efi grubx64.efi
8c5806e66bb5b052ebf860e1722474269cff3dde588610df21dbe8cf12c08390 shimx64.efi
546a71319c22da1d81879383c4c74be06d1c374bdecfafc9fcc80bd541802bfc grubx64.efi
random:~/devel/github.com/pesign/pesum$
```

This PR is on top of #82.

### CODE REVIEW

File: src/pesum.c [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+#include <prerror.h>
+
+#include "pesign.h"
+#include "pesign_standalone.h"
+
+static struct {
+	int flag;
+	const char *name;
+} flag_names[] = {
+	{DAEMONIZE, "daemonize"},
```
Comment (frozencemetery) [inline src/pesum.c:22]: flag_names and print_flag_name() don't seem to be used for anything.

File: src/pesum.c [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+		ctxp->verbose = verbose_cmd_line;
+
+		ctxp->hash = 1;
+		ctxp->infile = strdup(infile);
+		if (!ctxp->infile)
+			err(1, "Could not allocate memory");
+
+		rc = set_digest_parameters(ctxp->cms_ctx, digest_name);
+		int is_help  = strcmp(digest_name, "help") ? 0 : 1;
```
Comment (frozencemetery) [inline src/pesum.c:159]: Style: extra space before =.

**Review by frozencemetery — CHANGES_REQUESTED** [review]

File: src/pesum.c [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+				break;
+		}
+
+		pesign_context_free(ctxp);
+	}
+
+	poptFreeContext(optCon);
+
+	if (digest_name && digest_name != orig_digest_name)
+		free(digest_name);
```
[the remark this answers is not in the document]
Comment (vathpela) [reply src/pesum.c:184]: If it's not the same pointer as orig_digest_name, then it was allocated by popt.

File: src/pesum.c [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+{
+	for (int i = 0; flag_names[i].flag != FLAG_LIST_END; i++) {
+		if (flag_names[i].flag == flag)
+			fprintf(f, "%s ", flag_names[i].name);
+	}
+}
+
+static long *verbose;
+
+long verbosity(void)
```
[the remark this answers is not in the document]
Comment (vathpela) [reply src/pesum.c:48]: This is used in common code, but has to be defined in the front end.

File: src/pesum.c (hunk 1 above)
Comment (vathpela) [reply src/pesum.c:22]: This is used in common code, but has to be defined in the front end (for fairly dumb reasons in this case.) That's something that could stand to be cleaned up at some point.

File: src/pesum.c (hunk 2 above)
Comment (vathpela) [reply src/pesum.c:159]: Fixed.

File: src/pesum.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (vathpela) [reply src/pesum.c]: (And I've fixed the style issue.)

**Review by frozencemetery — APPROVED** [review]

**Pull request merged**